### DISCUSSION TITLE
What are the best practices for synchronizing Identity Service and StatefulChatClient?

### DISCUSSION THREAD

**Discussion by BrianUribe6** [Azure/communication-ui-library#3326] [Q&A]
Hi, I have a requirement that involves maintaining a mapping between each user and the chat threads they belong to. I know this information is already provided by the ChatClient and the ChatThreadClient, but because this data will be queried very often, I have persisted this on a service that will store and format it according to business requirements. 

The challenge, however, is keeping ACS state and this "Identity Service" in sync and I was wondering on what are the best practices for something like this?  I know a naive approach is listening to every event on the client ("threadCreated", "participantAdded", etc) and updating the Identity Service accordingly, but since the events are coming directly from the client this is unreliable.

How would I reliably keep ACS and my locally persisted data in sync?

**Comment by alkwa-msft** [comment]
created an issue from this discussion.
https://github.com/Azure/communication-ui-library/issues/3343
